peps/pep-0758.rst

```diff
@@ -0,0 +1,474 @@
+download URLs returned by PyPI's Simple API. In practice though, tools such as
+``pip`` and ``uv`` may have constrained lists of hosts they will allow
+downloads from, such as PyPI's own ``pythonhosted.org`` domain.
+
+In this case, such tools will need to relax those constraints, but the exact policy for
+this is left to the installer tools themselves. Any number of approaches could be
+implemented, such as downloading the ``.rim`` file and verifying the
+``EXTERNAL-HOSTING.json`` metadata, or simply trusting the external downloads for any
+wheel with a matching checksum.  They could also query PyPI for the project's organization
+owner and support URI before trusting the download. They could warn the user when
+externally hosted wheel files are encountered, and/or require the use of a command line
+option to enable additional download hosts. Any of these verification policies could be
+chosen in configuration files.
+
+Installer tools should also probably provide better error messages when
+externally hosted wheels cannot be downloaded, e.g. because a host is
+unreachable. As described above, such tools could query enough metadata from
+PyPI to provide clear and distinct error messages pointing users to the
+package's external hosting support email or issue tracker.
+
+Constraints for external hosting services
+=========================================
+
+The following constraints lead to reliable and compatible external wheel hosting services:
+
+- External wheels **MUST** be served over HTTPS, with a certificate signed by
+  `Mozilla's root certificate store <https://wiki.mozilla.org/CA>`__. This ensures
+  compatibility with `pip <https://pip.pypa.io/en/stable/topics/https-certificates/>`__
+  and `uv
+  <https://docs.astral.sh/uv/configuration/authentication/#custom-ca-certificates>`__.  At
+  the time of this writing, ``pip`` 24.2 on Python 3.10 or newer uses the system
+  certificate store in addition to the Mozilla store provided by the third party `certifi
+  <https://pypi.org/project/certifi/>`__ Python package. ``uv`` uses the Mozilla store
+  provided by the `webpki-roots <https://github.com/rustls/webpki-roots>`__ crate, but not
+  the system store unless the ``--native-tls`` flag is given [#fn1]_.  *The PyPI
+  administrators may modify this requirement in the future, but compatibility with popular
+  installers will not be compromised.*
+- External wheel hosts **SHOULD** use a content delivery network (`CDN
+  <https://en.wikipedia.org/wiki/Content_delivery_network>`__), just as PyPI does.
+- External wheel hosts **MUST** commit to a stable URL for all wheels they host.
+- Externally hosted wheels **MUST NOT** be removed from an external wheel host unless the
+  corresponding ``.rim`` file is deleted from PyPI first, and **MUST NOT** remove external
+  wheels for yanked releases.
+- Externally wheel hosts **SHOULD** respond to `HTTP HEAD`_ requests on the ``.whl`` URL
```

Comment:
This is already required by the HTTP spec, as long as the path exists :)
https://www.rfc-editor.org/rfc/rfc9110.html#section-9.1-9

Reply:
Of course! 😄